modules/reitit-spec/src/reitit/coercion/spec.cljc

```diff
@@ -67,20 +68,27 @@
   (reify coercion/Coercion
     (-get-name [_] :spec)
     (-get-options [_] opts)
-    (-get-apidocs [this _ {:keys [parameters responses] :as info}]
-      (cond-> (dissoc info :parameters :responses)
-              parameters (assoc
-                           ::swagger/parameters
-                           (into
-                             (empty parameters)
-                             (for [[k v] parameters]
-                               [k (coercion/-compile-model this v nil)])))
-              responses (assoc
-                          ::swagger/responses
-                          (into
-                            (empty responses)
-                            (for [[k response] responses]
-                              [k (update response :body #(coercion/-compile-model this % nil))])))))
+    (-get-apidocs [this type {:keys [parameters responses]}]
+      (condp = type
+        :swagger (swagger/swagger-spec
+                   (merge
+                     (if parameters
+                       {::swagger/parameters
+                        (into
+                          (empty parameters)
+                          (for [[k v] parameters]
+                            [k (coercion/-compile-model this v nil)]))})
+                     (if responses
+                       {::swagger/responses
+                        (into
+                          (empty responses)
+                          (for [[k response] responses
+                                :let [response (set/rename-keys response {:body :schema})]]
+                            [k (update response :schema #(coercion/-compile-model this % nil))]))})))
+        (throw
```

Comment:
This code looks exactly like the code for Schema, so would it make sense to de-duplicate it?

Reply:
Schema is not implemented yet, might look different in the end. Marked

`;; TODO: this looks identical to spec, refactor when schema is done.`